### DISCUSSION TITLE
Allow only patch updates for specific dependency, while using one branch for everything

### DISCUSSION THREAD

**Discussion by artem-nefedov** [renovatebot/renovate#12846] [Request Help]
Is it possible to allow only patch updates for a specific dependency (all other dependencies should have all types allowed), and also keep our current configuration of using a single branch for all updates?

For branch grouping, we have:
```
{
...
  "groupName": "all dependencies",
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "groupName": "all dependencies",
      "additionalBranchPrefix": ""
    },
...
```

To limit updates, I've tried adding the following to packageRules:

```
    {
      "matchPackageNames": ["my-dependency"],
      "minor": {"enabled": false},
      "major": {"enabled": false}
    }
```

However, instead of limiting updates to patch only, this makes it so "my-dependency" isn't updated at all.  In logs, I see that it grabs latest version (which is a minor update), instead of grabbing latest patch version relative to current version. Dependency source is "docker", in case that matters.

I'm using renovate bot version 29.8.2.

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#separateminorpatch

**Reply by artem-nefedov** [reply]
My question was if it's possible to apply this rule AND keep all updates in a single branch.
With `separateMinorPatch`, I will have different branches, which is not what I want.
